tests/test_fe_handlers.py

```diff
@@ -100,3 +100,72 @@ def test_usertokens(session, users, http_client, base_url):
             headers={'X-Grouper-User': user.username})
     assert resp.code == 200
     assert "Disabled token: myFoobarToken" in resp.body
+
+@pytest.mark.gen_test
+def test_request_emails(graph, groups, permissions, session, standard_graph, users, base_url,
+                        http_client):
+    tech = groups["tech-ops"]
+
+    tech.canjoin = "canask"
+    tech.add(session)
+    session.commit()
+
+    before_reqs = session.query(Request).count()
+
+    # REQUEST 1
+
+    # Explicitly requery because pulling from the users dict causes DetachedSessionErrors
+    user = session.query(User).filter_by(username="testuser@a.co").scalar()
+    fe_url = url(base_url, '/groups/{}/join'.format(tech.groupname))
+    resp = yield http_client.fetch(fe_url, method="POST",
+            body=urlencode({"reason": "Test Request Please Ignore", "member": "User: {}".format(user.name)}),
+            headers={'X-Grouper-User': user.username})
+    assert resp.code == 200
+
+    assert session.query(AsyncNotification).count() == 2, "Only approvers for the requested group should receive an email"
+    assert session.query(AsyncNotification).filter_by(email="zay@a.co").count() == 1, "Owners should receive exactly one email per canask request"
+    assert session.query(AsyncNotification).filter_by(email="figurehead@a.co").count() == 1, "NP-Owners should receive exactly one email per canask request"
+
+    assert session.query(Request).count() == before_reqs + 1, "There should only be one added request"
```

Comment:
these checks that just select count against `AsyncNotification` are a tad brittle (can't refactor by moving code blocks around).

better options are
1. calling `grouper.email_util.process_async_emails` directly (which returns a count of sent emails). or
2. doing something similar to `tests.test_permissions._get_unsent_and_mark_as_sent_emails` that takes a username.

(2) probably is more inline with what you're doing.